components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -0,0 +1,447 @@
+                            }));
+
+            if (StringUtils.isEmpty(webhookData.getId())) {
+                return null;
+            }
+            Integer webhookInternalId = getWebhookInternalId(webhookId, tenantId);
+            if (webhookInternalId != null) {
+                webhookData.setEventsSubscribed(listWebhookEvents(webhookInternalId));
+            }
+            return webhookData;
+        } catch (TransactionException | DataAccessException e) {
+            throw WebhookManagementExceptionHandler.handleServerException(
+                    ErrorMessage.ERROR_CODE_WEBHOOK_GET_ERROR, e, webhookId);
+        }
+    }
+
+    /**
+     * Delete a webhook subscription from the database.
+     *
+     * @param webhookId Webhook subscription ID.
+     * @param tenantId  Tenant ID.
+     * @throws WebhookMgtException If an error occurs while deleting the webhook subscription.
+     */
+    @Override
+    public void deleteWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                template.executeUpdate(Query.DELETE_WEBHOOK,
+                        preparedStatement -> {
+                            preparedStatement.setString(Column.UUID, webhookId);
+                            preparedStatement.setInt(Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw WebhookManagementExceptionHandler.handleServerException(
+                    ErrorMessage.ERROR_CODE_WEBHOOK_DELETE_ERROR, e, webhookId);
+        }
+    }
+
+    /**
+     * Get all webhooks for a tenant from the database.
+     *
+     * @param tenantId Tenant ID.
+     * @return List of webhook subscriptions.
+     * @throws WebhookMgtException If an error occurs while retrieving webhook subscriptions.
+     */
+    @Override
+    public List<Webhook> getWebhooks(int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        List<Webhook> webhooks;
+
+        try {
+            webhooks = jdbcTemplate.executeQuery(Query.LIST_ALL_WEBHOOKS,
+                    (resultSet, rowNumber) -> mapWebhook(resultSet),
+                    preparedStatement -> {
+                        preparedStatement.setInt(Column.TENANT_ID, tenantId);
+                    });
+
+            // Fetch events for each webhook
+            for (Webhook webhook : webhooks) {
+                Integer webhookInternalId = getWebhookInternalId(webhook.getId(), tenantId);
+                if (webhookInternalId != null) {
+                    webhook.setEventsSubscribed(listWebhookEvents(webhookInternalId));
+                }
+            }
+
+            return webhooks;
```

Comment:
In a webhook list scenario let's not pull up events. 
In a UI commonly in a list interface user will see only the name, endpoint and web hook status.
The api response can include a 'self' property that locates to the complete resource